### PR TITLE
feat(web.exchange): add dkim feature

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master` 
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-11724 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
I added dkim feature. I reuse the same logical structure of MX / SRV auto configure (with some legacy code).

## Related
:flags: Translation
- [x] TRANS-52610

### CODE REVIEW

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    this.GLOBAL_DKIM_STATUS =
+      navigation.currentActionData.constant.GLOBAL_DKIM_STATUS;
+
+    this.dkimGuideLink =
+      DKIM_CONFIGURATION_GUIDE[coreConfig.getUser().ovhSubsidiary] ||
+      DKIM_CONFIGURATION_GUIDE.DEFAULT;
+
+    this.loading = true;
+
+    this.services.ExchangeDomains.gettingDNSSettings(
```
Comment (darsene) [inline packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js:39]: to move to `$onInit` function

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        this.navigation.resetAction();
+        this.messaging.writeError(
+          this.$translate.instant(
+            'exchange_tab_domain_diagnostic_add_field_failure',
+          ),
+          failure,
+        );
+      });
+
+    $scope.configDkim = () => this.configDkim();
```
Comment (darsene) [inline packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js:61]:
```suggestion
    $scope.configDkim =  this.configDkim;
```

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  }
+
+  configDkim() {
+    this.services.ExchangeDomains.getDkimSelector(
+      this.$routerParams.organization,
+      this.$routerParams.productId,
+      this.domain.name,
+    )
+      .then((dkimSelectors) => {
+        const promise = dkimSelectors.map((dkimSelector, index) => {
```
Comment (darsene) [inline packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js:78]:
```suggestion
        const promises = dkimSelectors.map((dkimSelector, index) => {
```
Rename to plural to indicate it's an array

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            this.$routerParams.productId,
+            this.domain.name,
+            {
+              selectorName: dkimSelector,
+              autoEnableDKIM: index === 0,
+              configureDkim: true,
+            },
+          );
+        });
+        return this.services.$q.all(promise);
```
Comment (darsene) [inline packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js:90]:
```suggestion
        return this.services.$q.all(promises);
```

File: packages/manager/modules/exchange/src/domain/domain.controller.js [hunk 5]
```diff
@@ -96,6 +103,23 @@ export default class ExchangeTabDomainsCtrl {
       });
   }
 
+  dkimGlobalStatus({ dkim }) {
+    if (dkim.length === 0) {
```
Comment (darsene) [inline packages/manager/modules/exchange/src/domain/domain.controller.js:107]: Could `dkim` be null ?

**Review by darsene — CHANGES_REQUESTED** [review]

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js (hunk 1 above)
Comment (ghyenne) [reply packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js:39]: I'm at a place where we have legacy code.
The data-wizard is built during the constructor and the $onInit is played after it has been built, so button management is not good if I put it in an onInit (I've lost some hair :blush: ).

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js (hunk 2 above)
Comment (ghyenne) [reply packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js:61]: As for the previous comment, I need the function to be binded for the data-wizard before it is built.

File: packages/manager/modules/exchange/src/domain/domain.controller.js (hunk 5 above)
Comment (ghyenne) [reply packages/manager/modules/exchange/src/domain/domain.controller.js:107]: Not it's an array of object.

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js (hunk 4 above)
Comment (ghyenne) [reply packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js]: It's fixed

File: packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js (hunk 3 above)
Comment (ghyenne) [reply packages/manager/modules/exchange/src/domain/dkim-autoconfig/domain-dkim-autoconfig.controller.js]: It's fixed

**Review by darsene — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**